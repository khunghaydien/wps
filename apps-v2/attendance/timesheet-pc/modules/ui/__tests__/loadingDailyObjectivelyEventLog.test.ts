import snapshotDiff from 'snapshot-diff';

import reducer, {
  // @ts-ignore
  __get__,
  actions,
} from '../loadingDailyObjectivelyEventLog';

const ACTION_TYPE = __get__('ACTION_TYPE');
const initialState = __get__('initialState');

describe('reducer()', () => {
  test('@@init', () => {
    // @ts-ignore
    const next = reducer(undefined, { type: '@@INIT' });
    expect(snapshotDiff(undefined, next)).toMatchSnapshot();
  });
  test(ACTION_TYPE.START, () => {
    // Arrange
    const prev = initialState;
    // Act
    const next = reducer(prev, actions.start());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(ACTION_TYPE.FINISH, () => {
    // Arrange
    const prev = true;
    // Act
    const next = reducer(prev, actions.finish());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
});
