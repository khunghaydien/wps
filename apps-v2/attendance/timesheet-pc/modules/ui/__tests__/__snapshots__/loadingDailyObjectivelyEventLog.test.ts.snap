// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @@init 1`] = `
"Snapshot Diff:
  Comparing two different types of values. Expected undefined but received boolean."
`;

exports[`reducer() TIMESHEET-PC/UI/LOADING_DAILY_OBJECTIVELY_EVENT_LOG_DIALOG/FINISH 1`] = `
"Snapshot Diff:
- First value
+ Second value

- true
+ false"
`;

exports[`reducer() TIMESHEET-PC/UI/LOADING_DAILY_OBJECTIVELY_EVENT_LOG_DIALOG/START 1`] = `
"Snapshot Diff:
- First value
+ Second value

- false
+ true"
`;
